### PR TITLE
support npm install from npm-shrinkwrap.json in nodejs plugin

### PR DESCRIPTION
**Pull request by mvayngrib**

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by sergiusens** [comment]
Can we have at least a unit test for this?

File: snapcraft/plugins/nodejs.py [hunk 1]
```diff
@@ -116,7 +116,10 @@ def _npm_install(self):
         npm_install = ['npm', '--cache-min=Infinity', 'install', '--global']
         for pkg in self.options.node_packages:
             self.run(npm_install + [pkg])
-        if os.path.exists(os.path.join(self.builddir, 'package.json')):
+        if (
+            os.path.exists(os.path.join(self.builddir, 'package.json')) or
+            os.path.exists(os.path.join(self.builddir, 'npm-shrinkwrap.json'))
+        ):
```
Comment (sergiusens) [inline snapcraft/plugins/nodejs.py]: maybe nicer:
```python
json_files = ('package.json', 'npm-shrinkwrap.json')
if any((os.path.exists(os.path.join(self.builddir, f)) for f in json_files))
```

**Comment by mvayngrib** [comment]
@sergiusens sorry, i'm not a Python guy, and don't have time to dive in :)

**Comment by kyrofa** [comment]
@mvayngrib please at least log a bug so we know where you're coming from here.

**Pull request closed without merging**